lib/engines/hckinstall/hckinstall.rb

```diff
@@ -276,31 +274,54 @@ def run_second(client:)
     end
 
     def prepare_setup_scripts_config
-      kit_string = @platform['kit']
-      kit_type = kit_string[0..2]
-      kit_version = ''
-      kit_type == 'HCK' || kit_version = kit_string[3..]
+      kit_type, kit_version = parse_kit_info
 
       config = {
         kit_type:,
         hlk_kit_ver: kit_version
       }
 
-      unless @kit_info['download_url'].nil?
-        download_kit_installer(@kit_info['download_url'],
-                               "#{kit_type}#{kit_version}", @hck_setup_scripts_path)
+      @kit_path = find_kit(kit_type, kit_version)
+
+      log_kit_status(@kit_path, kit_type, kit_version)
+
+      unless @kit_info['download_url'].nil? || @kit_path
```

Comment:
`@kit_info['download_url'].nil?` is always `true` here when `@kit_path.nil?` so don't check it.